### ISSUE TITLE
ggml_new_tensor_impl: not enough space in the scratch memory

### ISSUE THREAD

**Issue by Belarrius1** [LostRuins/koboldcpp#50]
Every time I advance a little in my discussions, I crash with the following error:

`Processing Prompt [BLAS] (1024 / 1301 tokens)ggml_new_tensor_impl: not enough space in the scratch memory`

My RAM is only 40% loaded, my Max Token is 2048 etc... I don' understand

**Comment by LostRuins** [comment]
What model are you using? I previously increased the batch size for BLAS from 512 to 1024, but I will have to revert it if it's causing issues.

**Comment by Belarrius1** [comment]
I use LLaMA 30B in 4 bits

**Comment by LostRuins** [comment]
Can you please try newest version v1.6, which has a reduced batch size, and let me know if it's still crashing?

**Comment by Belarrius1** [comment]
Okay, so, I crash again

`Processing Prompt (1 / 1 tokens)
./bot.sh: line 1: 597004 Segmentation fault      sudo python3 koboldcpp.py models/30B/ggml-model-q4_0.bin --port 666 --host 0.0.0.0 --stream --psutil_set_threads
belaserver@belaserver:/home/llama/koboldcpp$
`

Here the content of my "bot.sh"

`sudo python3 koboldcpp.py models/30B/ggml-model-q4_0.bin --port 666 --host 0.0.0.0 --stream --psutil_set_threads
`

RAM Used -> 38% before crash (38% of 64GB)

I have 128 GB total of SWAP (64 GB per NvME) and 32GB of ZRAM

**Comment by Belarrius1** [comment]
Yes that's it. And sometimes after the crash, if I raise, I have a huge "BLAS" which does this, and the AI can respond with a message, then the message after it crashes, as if it had reached a maximum.

I will try with nommap

**Comment by Belarrius1** [comment]
After 26 answers, I crash, but this time for a different reason it seems

![image](https://user-images.githubusercontent.com/129199373/231832934-cbfc4ec0-e1a8-4c61-9b3a-2ddc0e382924.png)

`
Processing Prompt (1 / 1 tokens)
./bot.sh: line 1: 1106934 Segmentation fault      sudo python3 koboldcpp.py models/30B/ggml-model-q4_0.bin --port 666 --host 0.0.0.0 --stream --psutil_set_threads --nommap
`

It makes the BLAS and "processing prompt" for 18 tokens then crash again.

**Comment by LostRuins** [comment]
It does somehow seem like you are running out of memory somehow, perhaps the scratch buffers are too small for 30B? Do you get the same issue on the main llama.cpp repo?
